### PR TITLE
allow timeout disable

### LINKED ISSUE #52: Disable timeout
how do you disable it?

also, add a nice message when something times out to let the user there is a timeout argument that has been defaulted to 30

```bash
A request timedout. current timeout set to 30.
change timeout by LitServe(..., timeout=30)
```

### PR DESCRIPTION
**Pull request by aniketmaurya**
<details>
  <summary><b>Before submitting</b></summary>

- [x] Was this discussed/agreed via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/main/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

</details>

## What does this PR do?

Partially fixes #52 

* Allows to disable timeout by setting `LitServer(..., timeout=False)  # or timeout=-1`
* Log msg that request was timed out and to adjust by providing `timeout` argument

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in GitHub issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -355,6 +355,13 @@ server = LitServer(SimpleLitAPI(), accelerator="cuda", devices=4, timeout=30)
 
 This is useful to avoid requests queuing up beyond the ability of the server to respond.
 
+
+To disable the timeout for long-running tasks, set `timeout=False` or `timeout=-1`:
+
+```python
+server = LitServer(SimpleLitAPI(), accelerator="cuda", devices=4, timeout=False)
```
Comment (lantiga) [inline README.md:362]: Let's remove additional arguments and focus on the one we are introducing
```suggestion
server = LitServer(SimpleLitAPI(), timeout=False)
```

**Review by lantiga — CHANGES_REQUESTED** [review]
Looks to me like the middleware acts too early.

We need to timeout here:
https://github.com/Lightning-AI/litserve/pull/63/files#diff-9200e67a7e7d9ae3515b0d1ceb2cdc26e6575edf756f71287e9119a102515a50R395

if you’re hanging there waiting for too long AND your request hasn’t yet been popped from the queue, then we timeout

however if it has been popped from the queue, then we let it finish because it will compute anyway (we don’t cancel predictions that are being computed at this stage, we only remove the ones that are queued to prevent the queue to grow unbounded)

**Comment by aniketmaurya** [comment]
@lantiga updated the PR as per feedback. now the requests will timeout only if it was waiting long in the queue.

File: tests/test_simple.py [hunk 2]
```diff
@@ -84,12 +88,26 @@ def encode_response(self, output) -> Response:
...
-    with TestClient(server.app) as client:
-        response = client.post("/predict", json={"input": 4.0})
-        assert response.status_code == 504
+    async with LifespanManager(server.app) as manager, AsyncClient(app=manager.app, base_url="http://test") as ac:
+        response1 = ac.post("/predict", json={"input": 4.0})
+        response2 = ac.post("/predict", json={"input": 5.0})
+        response1, response2 = await asyncio.gather(response1, response2)
+        # first request blocks the second request in queue
+        # request only times out if it is in queue
+        assert response2.status_code == 504, "Server takes longer than specified timeout and request should timeout"
```
Comment (lantiga) [inline tests/test_simple.py:101]: I'd also add an assert for response1.status_code

**Review by lantiga — APPROVED** [review]

**Comment by lantiga** [comment]
Nice async trick @aniketmaurya

**Pull request merged**